addons/account/models/partner.py

```diff
@@ -437,8 +457,8 @@ def _get_company_currency(self):
     invoice_warn_msg = fields.Text('Message for Invoice')
     # Computed fields to order the partners as suppliers/customers according to the
     # amount of their generated incoming/outgoing account moves
-    supplier_rank = fields.Integer(default=0, copy=False)
-    customer_rank = fields.Integer(default=0, copy=False)
+    supplier_rank = fields.Integer(default=_default_supplier_rank, copy=False)
+    customer_rank = fields.Integer(default=_default_customer_rank, copy=False)
```

Comment:
I'm not sure what the difference with the `create` method implementation is.
Since `copy=False`, the field name should not be part of the create `vals_list`, so it should be gotten from the context too

Reply:
And if it isn't part of vals_list it gets it from defaults, which gets it from context.